### PR TITLE
Calendar updates

### PR DESCRIPTION
**Pull request by joshwooding**
- Update the Calendar stories
- Change the unselectable high emphasis to medium
- Add tooltip to previous and next month buttons
- Prevent a month only calendar wrapping to the next or previous year

### CODE REVIEW

**Comment by joshwooding** [comment]
Blocked by #261

**Review by origami-z — DISMISSED** [review]

File: packages/lab/src/calendar/internal/CalendarDay.tsx [hunk 1]
```diff
@@ -58,9 +60,10 @@ export const CalendarDay = forwardRef<HTMLButtonElement, CalendarDayProps>(
         {
           [withBaseName("hidden")]: hidden,
           [withBaseName("outOfRange")]: outOfRange,
-          [withBaseName("unselectableHighEmphasis")]: unselectable === "high",
           [withBaseName("today")]: today,
           [withBaseName("unselectableLowEmphasis")]: unselectable === "low",
+          [withBaseName("unselectableMediumEmphasis")]:
```
Comment (libertymayc) [inline packages/lab/src/calendar/internal/CalendarDay.tsx:65]: to match our conventions could we make this uitkEmphasisMedium/Low as usual, and selector .uitkCalendarDay.uitkEmphasisMedium/Low or .uitkCalendarDay-unselectable.uitkEmphasisMedium/Low ?

File: packages/lab/src/calendar/internal/CalendarDay.css [hunk 2]
```diff
@@ -1,18 +1,10 @@
...
-}
-
 /* TODO: Design need to align characteristics for CalendarDay */
 .uitkCalendarDay {
   --calendar-day-text-color: var(--uitk-text-primary-foreground);
   --calendar-day-background: var(--uitk-selectable-background);
   --calendar-day-background-hover: var(--uitk-selectable-background-hover);
   --calendar-day-text-color-hover: var(--uitk-selectable-foreground-hover);
-  --calendar-day-out-of-range-text-color: var(--uitk-text-secondary-foreground);
+  --calendar-day-out-of-range-text-color: var(--uitk-text-secondary-color-disabled);
```
Comment (libertymayc) [inline packages/lab/src/calendar/internal/CalendarDay.css:7]: might be best to make this token ..-outOfRange-.. as it's the 'type' part of the token (like 'blocked')

**Review by ZarkoPernar — APPROVED** [review]

File: packages/lab/src/calendar/internal/CalendarDay.css [hunk 3]
```diff
@@ -1,59 +1,53 @@
...
-
-  --calendar-day-selected-span-background: var(--uitk-selectable-background-hover);
-  --calendar-day-selected-span-text-color: var(--uitk-selectable-foreground);
-
-  --calendar-day-hovered-span-background: var(--uitk-selectable-background-hover);
-  --calendar-day-hovered-span-text-color: var(--uitk-selectable-foreground);
-
-  --calendar-day-hovered-offset-background: var(--uitk-selectable-background-hover);
-  --calendar-day-hovered-offset-text-color: var(--uitk-selectable-foreground);
+  --calendarDay-text-color: var(--uitk-text-primary-foreground);
```
Comment (libertymayc) [inline packages/lab/src/calendar/internal/CalendarDay.css:3]: we should keep this as --calendar-day as Calendar is the component and day is the subject - i've made this standard across components with a lint rule too in https://github.com/jpmorganchase/uitk/pull/335

File: packages/lab/src/calendar/internal/CalendarDay.css [hunk 4]
```diff
@@ -63,106 +57,105 @@
...
+  background: var(--calendarDay-background-hover);
+  color: var(--calendarDay-text-color-hover);
 }
 
-.uitkCalendarDay-unselectableHighEmphasis,
-.uitkCalendarDay-unselectableHighEmphasis:hover {
-  color: var(--calendar-day-blocked-text-color);
-  cursor: var(--calendar-day-blocked-cursor);
-  background: var(--calendar-day-blocked-background);
+.uitkCalendarDay-unselectable.uitkMediumEmphasis,
```
Comment (libertymayc) [inline packages/lab/src/calendar/internal/CalendarDay.css:73]: we use uitkEmphasisMedium in other components (switch medium / emphasis)

File: packages/lab/src/calendar/internal/CalendarDay.tsx [hunk 5]
```diff
@@ -56,9 +58,10 @@ export const CalendarDay = forwardRef<HTMLButtonElement, CalendarDayProps>(
         {
           [withBaseName("hidden")]: hidden,
           [withBaseName("outOfRange")]: outOfRange,
-          [withBaseName("unselectableHighEmphasis")]: unselectable === "high",
           [withBaseName("today")]: today,
-          [withBaseName("unselectableLowEmphasis")]: unselectable === "low",
+          [withBaseName("unselectable")]: !!unselectable,
+          uitkLowEmphasis: unselectable === "low",
```
Comment (libertymayc) [inline packages/lab/src/calendar/internal/CalendarDay.tsx:63]: are these needed if the classes don't do anything? (if so, also switch to uitkEmphasisLow here)

File: packages/lab/src/calendar/internal/CalendarDay.css (hunk 3 above)
Comment (joshwooding) [reply packages/lab/src/calendar/internal/CalendarDay.css:3]: Technically the component is CalendarDay? Or are we naming this inline with the public component

File: packages/lab/src/calendar/internal/CalendarDay.css (hunk 4 above)
Comment (libertymayc) [reply packages/lab/src/calendar/internal/CalendarDay.css:73]: also im curious, why is this an emphasis concept? as we usually have emphasis coming from characteristics and this is component specific

Comment (joshwooding) [reply packages/lab/src/calendar/internal/CalendarDay.css:73]: @bhoppers2008 would probably be better off answering this

File: packages/lab/src/calendar/internal/CalendarDay.tsx (hunk 5 above)
Comment (joshwooding) [reply packages/lab/src/calendar/internal/CalendarDay.tsx:63]: I think it's fine to keep them. Low emphasis is used for styling within CSS but medium isn't. We could remove medium but a red cross is added in medium but via the dom.

File: packages/lab/src/calendar/internal/CalendarDay.css [hunk 6]
```diff
@@ -23,37 +15,39 @@
...
   --calendar-day-size: var(--uitk-size-base);
-  --calendar-day-font-size: var(--uitk-text-fontSize);
+  --calendar-day-fontSize: var(--uitk-text-fontSize);
 }
 
 .uitkCalendarDay {
   width: var(--calendar-day-size);
   height: var(--calendar-day-size);
   color: var(--calendar-day-text-color);
   background-color: var(--calendar-day-background);
```
Comment (libertymayc) [inline packages/lab/src/calendar/internal/CalendarDay.css:49]: Use background as the attr here for more flexibility

File: packages/lab/src/calendar/internal/CalendarDay.css [hunk 7]
```diff
@@ -68,41 +62,40 @@
...
+.uitkCalendarDay-unselectable.uitkMediumEmphasis,
+.uitkCalendarDay-unselectable.uitkMediumEmphasis:hover {
   color: var(--calendar-day-blocked-text-color);
   cursor: var(--calendar-day-blocked-cursor);
   background: var(--calendar-day-blocked-background);
 }
 
 .uitkCalendarDay-today {
-  border: 1px solid var(--calendar-day-current-day-border-color);
+  border: 1px solid var(--calendar-day-currentDay-borderColor);
```
Comment (libertymayc) [inline packages/lab/src/calendar/internal/CalendarDay.css:81]: could use container border width and style here, out of scope of PR though

File: packages/lab/src/calendar/internal/CalendarDay.css (hunk 6 above)
Comment (libertymayc) [reply packages/lab/src/calendar/internal/CalendarDay.css:49]: (out of scope of PR though)

**Review by libertymayc — APPROVED** [review]

**Pull request merged**